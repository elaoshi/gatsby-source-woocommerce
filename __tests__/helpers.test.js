const { normaliseFieldName, processNode } = require("../helpers/");

describe("Helpers:", () => {
  it("should normalize fieldname", () => {
    expect(normaliseFieldName("products/categories")).toEqual(
      "productsCategories"
    );
  });

  it("should return correct data from processNode", () => {
    const __type = "wcProducts";
    const id = 1234;
    const cat_id = 23;

    const nodeWithOutType = {
      id,
      categories: [{ id: cat_id }],
      wordpress_id: id,
    };
    const node = {
      __type,
      ...nodeWithOutType,
    };
    const contentDigest = "digest_string";
    const createContentDigest = jest.fn(() => contentDigest);
    const processNodeResult = processNode(createContentDigest, node);

    expect(createContentDigest).toBeCalled();
    expect(processNodeResult).toEqual({
      ...nodeWithOutType,
      parent: null,
      children: [],
      internal: {
        type: __type,
        contentDigest,
      },
    });

    expect(processNodeResult).toMatchSnapshot({
      id: id,
      categories: [
        {
          id: cat_id,
          wordpress_id: cat_id,
        },
      ],
      wordpress_id: id,
      parent: expect.any(Object),
      children: expect.any(Array),
      internal: {
        type: __type,
        contentDigest: contentDigest,
      },
    });
  });
});
