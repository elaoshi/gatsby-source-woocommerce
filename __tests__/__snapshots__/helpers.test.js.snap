// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helpers: should return correct data from processNode 1`] = `
Object {
  "categories": Array [
    Object {
      "id": 23,
      "wordpress_id": 23,
    },
  ],
  "children": Any<Array>,
  "id": 1234,
  "internal": Object {
    "contentDigest": "digest_string",
    "type": "wcProducts",
  },
  "parent": Any<Object>,
  "wordpress_id": 1234,
}
`;
